### PR TITLE
[CatalogPromotion] refactor action type

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | 1.11
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php [hunk 1]
```diff
@@ -112,4 +90,19 @@ public function getBlockPrefix(): string
     {
         return 'sylius_catalog_promotion_action';
     }
+
+    private function addTypeToForm(FormEvent $event): void {
```
Comment (SirDomin) [inline src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php:94]:
```suggestion
    private function addTypeToForm(FormEvent $event): void 
    {
```

File: src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php [hunk 2]
```diff
@@ -48,18 +48,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
-
-                $form = $event->getForm();
-
-                $actionConfigurationType = $this->actionConfigurationTypes[$data->getType()];
-                $form->add('configuration', $actionConfigurationType, [
-                    'label' => false,
-                ]);
+               $this->addTypeToForm($event);
             })
             ->addEventListener(FormEvents::PRE_SUBMIT, function(FormEvent $event): void {
```
Comment (SirDomin) [inline src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php:53]: i guess the problem was with this EventListener as its kinda big, maybe we could move some logic to private method to understand what's going on?

File: src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php (hunk 1 above)
Comment (GSadee) [inline src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php:94]:
```suggestion
    private function addConfigurationTypeToForm(FormEvent $event): void
    {
```

File: src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php (hunk 2 above)
Comment (GSadee) [reply src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php:53]: @AdamKasp have you checked if all 3 listeners are needed?

Comment (AdamKasp) [reply src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php:53]: @SirDomin TBH I Don't see good way to exclude sth to private method,
@GSadee I've checked it :)

**Comment by GSadee** [comment]
The base of this pull-request was changed, you need fetch and reset your local branch 
if you want to add new commits to this pull request. **Reset before you pull, else commits
may become messed-up.**

Unless you added new commits (to this branch) locally that you did not push yet, 
execute `git fetch origin && git reset "event-listeners-CP-action-refactor"` to update your local branch.
 
Feel free to ask for assistance when you get stuck :+1:

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Adam! :1st_place_medal:

**Pull request merged**